mongodbatlas/resource_mongodbatlas_advanced_cluster.go

```diff
@@ -1079,6 +1141,38 @@ func replicationSpecsHashSet(v interface{}) int {
 	return schema.HashString(buf.String())
 }
 
+func getUpgradeRequest(d *schema.ResourceData) *matlas.Cluster {
+	if !d.HasChange("replication_specs") {
+		return nil
+	}
+
+	crs, nrs := d.GetChange("replication_specs")
+	cReplicationSpecs := expandAdvancedReplicationSpecs(crs.(*schema.Set).List())
+	nReplicationSpecs := expandAdvancedReplicationSpecs(nrs.(*schema.Set).List())
```

Comment:
[q] Could you give different names to these variables? The first time that I read this code I thought they were the same

Reply:
Yeah :facepalm: I put more emphasis on clarity of `ReplicationSpecs` than the differentiation between the two. These variables and others like it have been updated to `current{Specs|Region|Size}` and `updated{Specs|Region|Size}`